pkg/model/openstackmodel/firewall.go

```diff
@@ -36,113 +42,448 @@ type FirewallModelBuilder struct {
 
-						PortRangeMin: i(nodePortRange.Base),
-						PortRangeMax: i(nodePortRange.Base + nodePortRange.Size - 1),
+		nodePortRange, err := b.NodePortRange()
+		if err != nil {
+			return err
+		}
 
-						RemoteIPPrefix: s(nodePortAccess),
-					}
-					c.AddTask(t)
-				}
+		for _, protocol := range []string{IpProtocolTCP, IpProtocolUDP} {
+			nodePortRule := &openstacktasks.SecurityGroupRule{
+				Lifecycle:      b.Lifecycle,
+				Direction:      s(string(rules.DirIngress)),
+				Protocol:       s(protocol),
+				EtherType:      s(IPV4),
+				PortRangeMin:   i(nodePortRange.Base),
+				PortRangeMax:   i(nodePortRange.Base + nodePortRange.Size - 1),
+				RemoteIPPrefix: s(nodePortAccess),
 			}
-		} else if role == kops.InstanceGroupRoleMaster {
-			for _, apiAccess := range b.Cluster.Spec.NodePortAccess {
-				t := &openstacktasks.SecurityGroupRule{
-					Lifecycle: b.Lifecycle,
-					SecGroup:  sg,
-					Direction: s(DirectionIngress),
-					Protocol:  s(IpProtocolTCP),
-					EtherType: s(IPV4),
-
-					PortRangeMin: i(443),
-					PortRangeMax: i(443),
-
-					RemoteIPPrefix: s(apiAccess),
-				}
-				c.AddTask(t)
+			addDirectionalGroupRule(c, nodeSG, nil, nodePortRule)
+		}
+	}
+	return nil
+}
+
+// addHTTPSRules - Add rules to 443 access given the presence of a loadbalancer or not
+func (b *FirewallModelBuilder) addHTTPSRules(c *fi.ModelBuilderContext, sgMap map[string]*openstacktasks.SecurityGroup) error {
+
+	masterName := b.SecurityGroupName(kops.InstanceGroupRoleMaster)
+	nodeName := b.SecurityGroupName(kops.InstanceGroupRoleNode)
+	lbSGName := b.Cluster.Spec.MasterPublicName
+	lbSG := sgMap[lbSGName]
+	masterSG := sgMap[masterName]
+	nodeSG := sgMap[nodeName]
+
+	httpsIngress := &openstacktasks.SecurityGroupRule{
+		Lifecycle:    b.Lifecycle,
+		Direction:    s(string(rules.DirIngress)),
+		Protocol:     s(IpProtocolTCP),
+		EtherType:    s(IPV4),
+		PortRangeMin: i(443),
+		PortRangeMax: i(443),
+	}
+
+	//Allow all local communication for kubernetes.svc and to the api.internal lb/gossip for kubelet's
+	addDirectionalGroupRule(c, masterSG, nodeSG, httpsIngress)
+	addDirectionalGroupRule(c, masterSG, masterSG, httpsIngress)
+	addDirectionalGroupRule(c, nodeSG, masterSG, httpsIngress)
+	addDirectionalGroupRule(c, nodeSG, nodeSG, httpsIngress)
+
+	if b.UseLoadBalancerForAPI() {
+		//Allow API Access to the lb sg
+		for _, apiAccess := range b.Cluster.Spec.KubernetesAPIAccess {
+			addDirectionalGroupRule(c, lbSG, nil, &openstacktasks.SecurityGroupRule{
+				Lifecycle:      b.Lifecycle,
+				Direction:      s(string(rules.DirIngress)),
+				Protocol:       s(IpProtocolTCP),
+				EtherType:      s(IPV4),
+				PortRangeMin:   i(443),
+				PortRangeMax:   i(443),
+				RemoteIPPrefix: s(apiAccess),
+			})
+		}
+		//Allow masters ingress from the sg
+		addDirectionalGroupRule(c, masterSG, lbSG, httpsIngress)
+
+		//FIXME: Octavia port traffic appears to be denied though its port is in lbSG
+		if b.usesOctavia() {
```

Comment:
@zetaab I don't have an octavia openstack atm.  Would you be able to see if this fixes your issue?